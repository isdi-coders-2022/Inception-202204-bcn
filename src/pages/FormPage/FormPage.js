import Formulary from "../../components/Formulary/Formulary";
import FomrPageStyles from "./FormPage.styles";

const FormPage = () => {
  return (
    <FomrPageStyles>
      <Formulary />
    </FomrPageStyles>
  );
};

export default FormPage;
