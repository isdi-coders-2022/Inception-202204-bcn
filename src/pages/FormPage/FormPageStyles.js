import styled from "styled-components";

const FormPageStyles = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

export default FormPageStyles;
