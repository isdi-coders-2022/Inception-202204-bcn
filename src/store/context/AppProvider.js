import { useEffect, useReducer } from "react";
import { loadAlcoholicDrinksActionCreation } from "../actions/creador";
import appReducer from "../appReducer/appReducer";
import AppContext from "./AppContext";

const AppProvider = ({ children }) => {
  const dataBase = {
    alcoholicDrinks: [],
    nonAlcoholicDrinks: [],
    mostPopularDrinks: [],
    latestDrinks: [],
  };

  const [drinksData, dispatch] = useReducer(appReducer, dataBase);

  useEffect(() => {
    (async () => {
      const response = await fetch(
        `https://www.thecocktaildb.com/api/json/v1/1/filter.php?a=Alcoholic`
      );
      let drinksData = await response.json();

      dispatch(loadAlcoholicDrinksActionCreation(drinksData.drinks));
    })();
  }, []);

  return (
    <AppContext.Provider value={{ drinksData, dispatch }}>
      {children}
    </AppContext.Provider>
  );
};

export default AppProvider;
