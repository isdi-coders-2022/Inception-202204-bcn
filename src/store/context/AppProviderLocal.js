import { useEffect, useReducer } from "react";
import { loadUserDrinksActionCreation } from "../actions/creador";
import appReducer from "../appReducer/appReducer";
import AppContextLocal from "./AppContextLocal";

const AppProviderLocal = ({ children }) => {
  const dataBase = {
    drinks: [],
  };

  let [drinksData, dispatch] = useReducer(appReducer, dataBase);

  useEffect(() => {
    (async () => {
      const response = await fetch(`https://drinks-api.onrender.com/drinks`);
      let drinksData = await response.json();

      dispatch(loadUserDrinksActionCreation(drinksData));
    })();
  }, []);

  return (
    <AppContextLocal.Provider value={{ drinksData, dispatch }}>
      {children}
    </AppContextLocal.Provider>
  );
};

export default AppProviderLocal;
