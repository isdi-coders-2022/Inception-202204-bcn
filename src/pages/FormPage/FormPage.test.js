import React from "react";
import FormPage from "./FormPage";
import TestRenderer from "react-test-renderer";
import "@testing-library/jest-dom";
import AppProvider from "../../store/context/AppProvider";
import AppProviderLocal from "../../store/context/AppProviderLocal";

describe("Given FormPage component", () => {
  describe("When rendered", () => {
    test("Then it should always match this snapshot", () => {
      const formularyPage = TestRenderer.create(
        <AppProvider>
          <AppProviderLocal>
            <FormPage />
          </AppProviderLocal>
        </AppProvider>
      );
      expect(formularyPage).toMatchSnapshot();
    });
  });
});
