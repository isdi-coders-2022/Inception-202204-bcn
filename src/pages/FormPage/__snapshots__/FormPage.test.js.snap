// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given FormPage component When rendered Then it should always match this snapshot 1`] = `
<div
  className="sc-gsnTZi ToOMS"
>
  <form
    className="sc-bczRLJ hqXwgW"
    onSubmit={[Function]}
  >
    <label>
      Cocktail name
      <br />
      <input
        autoComplete="off"
        name="name"
        required={true}
        type="text"
      />
    </label>
    <label>
      Type of drink
      <br />
      <select
        name="category"
      >
        <option
          value="Beer"
        >
          Beer
        </option>
        <option
          value="Cocktail"
        >
          Cocktail
        </option>
        <option
          value="Coffe / Tea"
        >
          Coffe / Tea
        </option>
        <option
          value="Ordinary Drink"
        >
          Ordinary Drink
        </option>
        <option
          value="Shake"
        >
          Shake
        </option>
        <option
          value="Shot"
        >
          Shot
        </option>
        <option
          value="Soft Drink"
        >
          Soft Drink
        </option>
      </select>
    </label>
    <label>
      Type
      <br />
      <select
        name="alcoholic"
      >
        <option
          value="Alcoholic"
        >
          Alcoholic
        </option>
        <option
          value="Non-Alcoholic"
        >
          Non-Alcoholic
        </option>
      </select>
    </label>
    <label>
      Type of glass
      <br />
      <select
        name="glass"
      >
        <option
          value="Cocktail Glass"
        >
          Cocktail Glass
        </option>
        <option
          value="Collins Glass"
        >
          Collins Glass
        </option>
        <option
          value="Copper Mug"
        >
          Copper Mug
        </option>
        <option
          value="Highball Glass"
        >
          Highball Glass
        </option>
        <option
          value="Irish Coffe Cup"
        >
          Irish Coffe Cup
        </option>
        <option
          value="Old-Fashioned Glass"
        >
          Old-Fashioned Glass
        </option>
        <option
          value="Whiskey glass"
        >
          Whiskey glass
        </option>
      </select>
    </label>
    <label>
      Preparation
      <br />
      <textarea
        autoComplete="off"
        cols="30"
        name="instructions"
        placeholder="Cocktail Instructions"
        required={true}
        rows="7"
      />
    </label>
    <label>
      Ingredient 1
      <br />
      <select
        name="ingredient1"
      >
        <option />
        <option
          value="Angostura"
        >
          Angostura
        </option>
        <option
          value="Cognac"
        >
          Cognac
        </option>
        <option
          value="Dark rum"
        >
          Dark rum
        </option>
        <option
          value="Light rum"
        >
          Light rum
        </option>
        <option
          value="Gin"
        >
          Gin
        </option>
        <option
          value="Tequila"
        >
          Tequila
        </option>
        <option
          value="Triple Sec"
        >
          Triple Sec
        </option>
        <option
          value="Vodka"
        >
          Vodka
        </option>
      </select>
    </label>
    <label>
      Ingredient 2
      <br />
      <select
        name="ingredient2"
      >
        <option />
        <option
          value="Sweet Vermouth"
        >
          Sweet Vermouth
        </option>
        <option
          value="Dry Vermouth"
        >
          Dry Vermouth
        </option>
        <option
          value="Champagne"
        >
          Champagne
        </option>
        <option
          value="Bourbon"
        >
          Bourbon
        </option>
        <option
          value="Kahlua"
        >
          Kahlua
        </option>
        <option
          value="Irish whiskey"
        >
          Irish whiskey
        </option>
        <option
          value="Sherry"
        >
          Sherry
        </option>
      </select>
    </label>
    <label>
      Ingredient 3 
      <br />
      <select
        name="ingredient3"
      >
        <option />
        <option
          value="Grenadine"
        >
          Grenadine
        </option>
        <option
          value="Lemon juice"
        >
          Lemon juice
        </option>
        <option
          value="Lime juice"
        >
          Lime juice
        </option>
        <option
          value="Orange juice"
        >
          Orange juice
        </option>
        <option
          value="Pineapple juice"
        >
          Pineapple juice
        </option>
        <option
          value="Soda water"
        >
          Soda water
        </option>
        <option
          value="Sprite"
        >
          Sprite
        </option>
        <option
          value="Sugar syrup"
        >
          Sugar syrup
        </option>
        <option
          value="Sweet Vermouth"
        >
          Sweet Vermouth
        </option>
        <option
          value="Tonic water"
        >
          Tonic water
        </option>
      </select>
    </label>
    <label>
      Ingredient 4
      <br />
      <select
        name="ingredient4"
      >
        <option />
        <option
          value="Apple juice"
        >
          Apple juice
        </option>
        <option
          value="Milk"
        >
          Milk
        </option>
        <option
          value="Chocolate syrup"
        >
          Chocolate syrup
        </option>
        <option
          value="Tomato juice"
        >
          Tomato juice
        </option>
        <option
          value="Coffee"
        >
          Coffee
        </option>
        <option
          value="Cranberry juice"
        >
          Cranberry juice
        </option>
        <option
          value="Grape juice"
        >
          Grape juice
        </option>
        <option
          value="7-Up"
        >
          7-Up
        </option>
        <option
          value="Tea"
        >
          Tea
        </option>
        <option
          value="Creme de Cacao"
        >
          Creme de Cacao
        </option>
      </select>
    </label>
    <label>
      Ingredient 5
      <br />
      <select
        name="ingredient5"
      >
        <option />
        <option
          value="Cream"
        >
          Cream
        </option>
        <option
          value="Cubes ice"
        >
          Cubes ice
        </option>
        <option
          value="Egg"
        >
          Egg
        </option>
        <option
          value="Lemon peel"
        >
          Lemon peel
        </option>
        <option
          value="Lime"
        >
          Lime
        </option>
        <option
          value="Mango"
        >
          Mango
        </option>
        <option
          value="Mint"
        >
          Mint
        </option>
        <option
          value="Vanilla extract"
        >
          Vanilla extract
        </option>
      </select>
    </label>
    <label>
      Url Image
      <br />
      <input
        autoComplete="off"
        name="urlImage"
        type="text"
      />
    </label>
    <button
      type="submit"
      value="Save Cocktail"
    >
      Save Cocktail
    </button>
  </form>
</div>
`;
